gateway-js/src/index.ts

```diff
@@ -467,6 +445,102 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
+  private async updateCsdl(result: CsdlUpdate): Promise<void> {
+    if (result.id === this.compositionId) {
+      this.logger.debug('No change in composition since last check.');
+      return;
+    }
+
+    const previousSchema = this.schema;
+    const previousCsdl = this.parsedCsdl;
+    const previousCompositionId = this.compositionId;
+
+    if (previousSchema) {
+      this.logger.info('Updated CSDL was found.');
+    }
+
+    await this.maybePerformServiceHealthCheck(result);
+
+    this.compositionId = result.id;
+    this.parsedCsdl = parse(result.csdl);
+
+    if (this.queryPlanStore) this.queryPlanStore.flush();
+
+    const { schema, composedSdl } = this.createSchema({
```

Comment:
I think this call and the one in updateServiceDefs could just call the underlying function being called here directly...

Reply:
Agree! 544ddafa